To-do list/Controllers/LoginViewController.swift

```diff
@@ -23,29 +26,76 @@ class LoginViewController: UIViewController {
+            return "Password must have at least 8 characters"
+        }
         return nil
-    }
+    }    
     
-    @IBAction func signInButton(_ sender: UIButton) {        
+    private func validateCredentials() -> Bool {
         let loginMessage = validateEmail()
         let passMessage = validatePass()
         
-        if loginMessage == nil && passMessage == nil {
-            let taskListVC = TaskListViewController.loadFromStoryboard(type: TaskListViewController.self)
-            self.navigationController?.pushViewController(taskListVC, animated: true)
-        } else {
+        if loginMessage != nil || passMessage != nil {
             var message = String()
             if let loginMessage = loginMessage {
                 message = loginMessage
             } else if let passMessage = passMessage {
                 message = passMessage
             }
             Alert.showBasic(title: message, vc: self)
+            return false
+        } else {
+            return true
+        }
+    }
+    
+    @IBAction func signInButton(_ sender: UIButton) {
+        if validateCredentials() {
+            guard let email = loginTextField.text,
+                  let password = passwordTextField.text else {return}
+            
+            authApiHelper.executeLoginRequest(email: email, password: password) { result in
+                var message = String()
+                if let result = result {
+                    if result.token != nil {
+                        APIHelper.token = result.token!
+                            let taskListVC = TaskListViewController.loadFromStoryboard(type: TaskListViewController.self)
+                        self.navigationController?.pushViewController(taskListVC,
+                                                                           animated: true)
+                    } else {
+                        message = "User doesn't exist with provided email and passsword"
+                    }
+                } else {
+                    message = "Something went wrong. Please try again."
+                }
+                if !message.isEmpty {
+                    Alert.showBasic(title: message, vc: self)
+                }
+            }
         }
     }
     
     @IBAction func registerButton(_ sender: UIButton) {
-        
+        if validateCredentials() {
+            guard let email = loginTextField.text,
+                  let password = passwordTextField.text else {return}
+            
+            authApiHelper.executeRegisterRequest(email: email, password: password) { result in
+                var message = String()
+                if let result = result {
+                    if result.status == "ok"{
+                        message = "Registration successfull"
+                    } else {
+                        message = "This email alredy exist"
+                    }
+                } else {
+                    message = "Something went wrong. Please try again."
```

Comment:
This text is used in lots of places. You can move it to some Constants enum or struct to avoid code duplication and improve the code maintainability. Imagine the case when you will need to update this text because of an updated design or new requirements. You will need to update the code in many places in your current implementation.

https://github.com/kodecocodes/swift-style-guide#constants - enum example

Reply:
Enum is added, thank you for useful link.